File: simplypdf/src/main/java/com/wwdablu/soumya/simplypdf/composers/models/ShapeProperties.java
package com.wwdablu.soumya.simplypdf.composers.models;

import com.google.gson.annotations.SerializedName;
import com.wwdablu.soumya.simplypdf.composers.Composer;
import com.wwdablu.soumya.simplypdf.composers.ShapeComposer;
import com.wwdablu.soumya.simplypdf.composers.UnitComposer;

public class ShapeProperties extends UnitComposer.Properties {

    @SerializedName("linecolor")
    public String lineColor;

    @SerializedName("linewidth")
    public int lineWidth;

    @SerializedName("shouldfill")
    public boolean shouldFill;

    public Composer.Alignment alignment;

    public ShapeProperties() {
        lineColor = "#000000";
        lineWidth = 1;
        shouldFill = false;
        alignment = Composer.Alignment.LEFT;
    }

    @Override
    public String getPropId() {
        return ShapeComposer.class.getName() + "Properties";
    }
}

File: simplypdf/src/main/java/com/wwdablu/soumya/simplypdf/composers/models/ShapeProperties.kt
package com.wwdablu.soumya.simplypdf.composers.models

import com.google.gson.annotations.SerializedName
import com.wwdablu.soumya.simplypdf.composers.Composer
import com.wwdablu.soumya.simplypdf.composers.ShapeComposer
import com.wwdablu.soumya.simplypdf.composers.UnitComposer

class ShapeProperties : UnitComposer.Properties() {

    @kotlin.jvm.JvmField
    @SerializedName("linecolor")
    var lineColor = "#000000"

    @kotlin.jvm.JvmField
    @SerializedName("linewidth")
    var lineWidth = 1

    @kotlin.jvm.JvmField
    @SerializedName("shouldfill")
    var shouldFill = false

    @kotlin.jvm.JvmField
    var alignment: Composer.Alignment = Composer.Alignment.START

    override val propId = ShapeComposer::class.java.name + "Properties"
}